### PR TITLE
[WFLY-13353] Galleon layer for the Pojo subsystem

### PR DESCRIPTION
**Pull request by yersan**
Jira issue: https://issues.redhat.com/browse/WFLY-13353

### CODE REVIEW

**Comment by darranl** [comment]
For all new layers I think we need to make sure security is considered in the requirements,

We don't want to accidentally leave anything unsecured by default but ideally if layers can avoid a dependency on any security layers that would be preferred.

Failing that for the new layers if security dependencies are required we should be preferring elytron with an option to select legacy security.  At some point in the future all legacy security related layers will be removed.

Can we please add a check for these questions in the requirements?

File: pojo/WFLY-13353_galleon_layer_for_pojo_subsystem.adoc [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+=== Testing By
+
+* [X] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+WildFly
```
Comment (fabiobrz) [inline pojo/WFLY-13353_galleon_layer_for_pojo_subsystem.adoc:63]: Isn't WildFly core affected as well? Can you add a link for each affected project?

**Review by fabiobrz — COMMENTED** [review]
Hi @yersan - thanks for this proposal. I've left just one comment, otherwise the AD LGTM. Let me know what you think

File: pojo/WFLY-13353_galleon_layer_for_pojo_subsystem.adoc [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+. Testing the Galleon layer provisioning. This testing is done by https://github.com/wildfly/wildfly/blob/master/testsuite/layers/src/test/java/org/jboss/as/test/layers/LayersTestCase.java[LayersTestCase]. The testsuite will be modified to add a new server provisioned with this layer in isolation and with this layer combined with all the layers. For each kind of provisioning, this test does the following:
+
+.. Verifies the provisioned modules are the expected ones.
+.. Verifies the provisioned server starts successfully.
+
+. Execution of WildFly tests related to the feature provisioned by this layer. Reuse the existing tests available on the WildFly test suite, which are directly testing this layer functionalities, and execute them on a server installation provisioned with this layer.
+
+== Community Documentation
+
+Community documentation plan is adding the layer to https://docs.wildfly.org/19/Admin_Guide.html#wildfly-galleon-layers[WildFly Galleon layers] in the section it belongs to.
```
Comment (fabiobrz) [inline pojo/WFLY-13353_galleon_layer_for_pojo_subsystem.adoc:96]: s/19/20/

**Review by fabiobrz — CHANGES_REQUESTED** [review]
Hi @yersan - noticed one minor typo too.

File: pojo/WFLY-13353_galleon_layer_for_pojo_subsystem.adoc (hunk 1 above)
Comment (yersan) [reply pojo/WFLY-13353_galleon_layer_for_pojo_subsystem.adoc:63]: The `pojo` subsystem is on WildFly not in WildFly core, so only WildFly is affected by this RFE.

**Review by fabiobrz — APPROVED** [review]

**Pull request merged**